src/pnm/encoder.rs

```diff
@@ -326,43 +329,38 @@ impl<'a> CheckedDimensions<'a> {
     // the comination is bogus (e.g. combining Pixmap and Palette) but allows uncertain
     // combinations (basically a ArbitraryTuplType::Custom with any color of fitting depth).
     fn check_header_color(self, color: ColorType) -> io::Result<CheckedHeaderColor<'a>> {
-        let components = match color {
-            ColorType::Gray(_) => 1,
-            ColorType::GrayA(_) => 2,
-            ColorType::Palette(_) | ColorType::RGB(_) | ColorType::BGR(_)=> 3,
-            ColorType::RGBA(_) | ColorType::BGRA(_) => 4,
-        };
+        let components = num_components(color) as u32;
 
         match *self.unchecked.header {
             PNMHeader {
                 decoded: HeaderRecord::Bitmap(_),
                 ..
             } => match color {
-                ColorType::Gray(_) => (),
+                ColorType::L1 | ColorType::L8 | ColorType::L16 => (),
                 _ => {
                     return Err(io::Error::new(
                         io::ErrorKind::InvalidInput,
-                        "PBM format only support ColorType::Gray",
+                        "PBM format only support ColorType::L",
```

Comment:
That's a rather odd way to put it. `::{L1, L8, L16}` would be clearer.

Reply:
Just saying "luma colortypes" is maybe simpler?

Reply:
I mean since I guess errors like this can end up being user-facing in some applications, and thus referencing the specific enum  could look sort of strange in some cases.